File: Python/381A: Sereja and Dima.py
def solve():
    n = int(input())
    cards = list(map(int, input().split()))
    
    left = 0
    right = n - 1
    
    sereja_score = 0
    dima_score = 0
    
    # Sereja starts (turn 0 is even)
    for turn in range(n):
        # Determine which card is larger
        if cards[left] > cards[right]:
            chosen_card = cards[left]
            left += 1
        else:
            chosen_card = cards[right]
            right -= 1
            
        # Assign points based on whose turn it is
        if turn % 2 == 0:
            sereja_score += chosen_card
        else:
            dima_score += chosen_card
            
    print(f"{sereja_score} {dima_score}")

solve()

File: CPP/381A: Sereja and Dima.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int n;
    cin >> n;
    
    vector<int> cards(n);
    for (int i = 0; i < n; i++) {
        cin >> cards[i];
    }
    
    int left = 0;
    int right = n - 1;
    int sereja = 0;
    int dima = 0;
    bool sereja_turn = true;
    
    while (left <= right) {
        if (cards[left] > cards[right]) {
            if (sereja_turn) sereja += cards[left];
            else dima += cards[left];
            left++;
        } else {
            if (sereja_turn) sereja += cards[right];
            else dima += cards[right];
            right--;
        }
        // Switch turns
        sereja_turn = !sereja_turn;
    }
    
    cout << sereja << " " << dima << endl;
    
    return 0;
}
